src/index.js

```diff
@@ -20,175 +20,131 @@ import keyboardNavigation from './tree-manager/keyboardNavigation'
 import './index.css'
 import { getAriaLabel } from './a11y'
 
-class DropdownTreeSelect extends Component {
-  static propTypes = {
-    data: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
-    clearSearchOnChange: PropTypes.bool,
-    keepTreeOnSearch: PropTypes.bool,
-    keepChildrenOnSearch: PropTypes.bool,
-    keepOpenOnSelect: PropTypes.bool,
-    texts: PropTypes.shape({
-      placeholder: PropTypes.string,
-      noMatches: PropTypes.string,
-      label: PropTypes.string,
-      labelRemove: PropTypes.string,
-    }),
-    showDropdown: PropTypes.oneOf(['default', 'initial', 'always']),
-    className: PropTypes.string,
-    onChange: PropTypes.func,
-    onAction: PropTypes.func,
-    onNodeToggle: PropTypes.func,
-    onFocus: PropTypes.func,
-    onBlur: PropTypes.func,
-    mode: PropTypes.oneOf(['multiSelect', 'simpleSelect', 'radioSelect', 'hierarchical']),
-    showPartiallySelected: PropTypes.bool,
-    disabled: PropTypes.bool,
-    readOnly: PropTypes.bool,
-    id: PropTypes.string,
-    searchPredicate: PropTypes.func,
-    inlineSearchInput: PropTypes.bool,
-  }
-
-  static defaultProps = {
-    onAction: () => {},
-    onFocus: () => {},
-    onBlur: () => {},
-    onChange: () => {},
-    texts: {},
-    showDropdown: 'default',
-    inlineSearchInput: false,
-  }
-
-  constructor(props) {
-    super(props)
-    this.state = {
-      searchModeOn: false,
-      currentFocus: undefined,
+const DropdownTreeSelect = ({
+  id: idProp,
+  data,
+  mode,
+  showDropdown: showDropdownProp,
+  showPartiallySelected,
+  searchPredicate,
+  className,
+  disabled,
+  readOnly,
+  texts,
+  inlineSearchInput,
+  tabIndex,
+  keepTreeOnSearch,
+  keepChildrenOnSearch,
+  onAction: onActionProp,
+  onChange,
+  onFocus,
+  onBlur,
+  onNodeToggle: onNodeToggleProp,
+  keepOpenOnSelect,
+  clearSearchOnChange,
+}) => {
+  const [state, setState] = useState({ searchModeOn: false })
+  const clientId = useMemo(() => idProp || clientIdGenerator.get(), [idProp])
+  const treeManager = useRef()
+  const searchInputRef = useRef()
+  const nodeRef = useRef()
+  const keepDropdownActive = useRef(false)
+  const callbackRef = useRef()
+
+  const setStateWithCallback = (newState, callback) => {
```

Comment:
I created it to emulate the callback arg in the old `setState`. In the future, this component can be refactored to not need it.